content/guides/code-splitting.md

```diff
@@ -1,41 +1,174 @@
 ---
 title: Code Splitting
-sort: 14
+sort: 9
 contributors:
   - pksjce
   - pastelsky
   - simon04
+  - jonwheeler
+  - johnstew
+  - shinxi
+  - tomtasche
+  - levy9527
+  - rahulcs
+  - chrisVillanueva
+  - rafde
+  - bartushek
+  - shaunwallace
+  - skipjack
+  - jakearchibald
+  - TheDutchCoder
 ---
 
-Code splitting is one of the most compelling features of webpack. It allows you to split your code into various bundles which you can then load on demand — like when a user navigates to a matching route, or on an event from the user. This allows for smaller bundles, and allows you to control resource load prioritization, which if used correctly, can have a major impact on your application load time.
+Code splitting is one of the most compelling features of webpack. This feature allows you to split your code into various bundles which can then be loaded on demand or in parallel. It can be used to achieve smaller bundles and control resource load prioritization which, if used correctly, can have a major impact on load time.
 
-There are mainly two kinds of code splitting that can be accomplished with webpack:
+There are three general approaches to code splitting available:
 
+- Entry Points: Manually split code using [`entry`](/configuration/entry-context) configuration.
+- Plugins & Loaders: Certain plugins and loaders can be used to split code in a variety of different ways.
+- Dynamic Imports: This method allows splitting code via inline function calls within modules.
 
-## Resource Splitting
 
-Vendor and CSS code splitting methods help with both caching and parallel loading.
+## Entry Points
 
-### Vendor Code Splitting
+This is by far the easiest, and most intuitive, way to split code. However, it is more manual and has a some pitfalls we will go over. Let's take a look at how we might split the core libraries and frameworks an application uses from the actual source code:
 
-A typical application can depend on many third party libraries for framework/functionality needs. Unlike application code, code present in these libraries does not change often.
+__webpack.config.js__
```

Comment:
You mean do the file diffs we did in other guides? Yeah we should add those here for consistency as well.